website/server/models/task.js

```diff
@@ -25,6 +26,13 @@ export let TaskSchema = new Schema({
   type: {type: String, enum: tasksTypes, required: true, default: tasksTypes[0]},
   text: {type: String, required: true},
   notes: {type: String, default: ''},
+  shortName: {
+    type: String,
+    validate: [function validateShortName (val) {
+      return Boolean(this.userId) && !validator.isUUID(val) && val.match(SHORT_NAME_VALIDATION_REGEX);
+    },
+    'Task short names can only contain alphanumeric characters, underscores and dashses. They must be owned by a user.'],
```

Comment:
Oh and it's probably worth stating that they can't be valid UUIDs. If you already have scripts set up for old tasks using a UUID, you might try to copy the UUID to the short name field, and then be confused about why it doesn't work because it does match the "alphanum, underscore, dash" restriction.